tests/e2e/aws_e2e.py

```diff
@@ -106,9 +109,48 @@ def test_end_to_end_other_volume(self):
         self.aws.ResourceApi(EC2_SERVICE).Volume(other_volume_copy.volume_id))
     self.assertEqual(self.volumes[-1].volume_id, other_volume_copy.volume_id)
 
+  def test_end_to_end_vm(self):
+    """End to end test on AWS.
+
+    This tests that an analysis VM is correctly created and that a volume
+        passed to the attach_volume parameter is correctly attached.
+    """
+
+    volume_to_attach = aws.CreateVolumeCopy(
+        self.zone,
+        volume_id=self.volume_to_forensic)
+    self.volumes.append(volume_to_attach)
+    # Create and start the analysis VM and attach the boot volume
+    self.analysis_vm, _ = aws.StartAnalysisVm(
+        self.analysis_vm_name,
+        self.zone,
+        10,
+        4,
+        attach_volume=volume_to_attach,
+        device_name='/dev/sdp'
+    )
+
+    # The forensic instance should be live in the analysis AWS account and
+    # the volume should be attached
+    instance = self.aws.ResourceApi(EC2_SERVICE).Instance(
+        self.analysis_vm.instance_id)
+    self.assertEqual(instance.instance_id, self.analysis_vm.instance_id)
+    for volume in instance.volumes.all():
+      if volume.volume_id == volume_to_attach.volume_id:
+        return
```

Comment:
What you want here is `assertIn()` https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn